File: java/parser/src/main/java/gov/lbl/parser/parser/Component_declaration1Visitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Comment;
import gov.lbl.parser.domain.Component_declaration1;
import gov.lbl.parser.domain.Declaration;

public class Component_declaration1Visitor extends modelicaBaseVisitor<Component_declaration1> {
    @Override
    public Component_declaration1 visitComponent_declaration1(modelicaParser.Component_declaration1Context ctx) {

        DeclarationVisitor declarationVisitor = new DeclarationVisitor();
        Declaration declaration = ctx.declaration() == null ? null : ctx.declaration().accept(declarationVisitor);

        CommentVisitor commentVisitor = new CommentVisitor();
        Comment comment = ctx.comment() == null ? null : ctx.comment().accept(commentVisitor);
        
        return new Component_declaration1(declaration, comment);
    }
}

File: jsParser/parser/Component_declaration1Visitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Component_declaration1 = require('../domain/Component_declaration1')

const DeclarationVisitor = require('./DeclarationVisitor')
const CommentVisitor = require('./CommentVisitor')

class Component_declaration1Visitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitComponent_declaration1 (ctx) {
    let declaration = null
    let comment = null

    if (ctx.declaration()) {
      const declarationVisitor = new DeclarationVisitor.DeclarationVisitor()
      declaration = declarationVisitor.visitDeclaration(ctx.declaration())
    }
    if (ctx.comment()) {
      const commentVisitor = new CommentVisitor.CommentVisitor()
      comment = commentVisitor.visitComment(ctx.comment())
    }

    return new Component_declaration1.Component_declaration1(declaration, comment)
  }
};

Component_declaration1Visitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitComponent_declaration = this.visitComponent_declaration
exports.Component_declaration1Visitor = Component_declaration1Visitor
